sklearn/preprocessing/data.py

```diff
@@ -251,19 +264,53 @@ def fit(self, X, y=None):
             The data used to compute the per-feature minimum and maximum
             used for later scaling along the features axis.
         """
-        X = check_array(X, copy=self.copy, ensure_2d=False, warn_on_dtype=True,
-                        estimator=self, dtype=FLOAT_DTYPES)
+        return self.partial_fit(X, y)
+
+    def partial_fit(self, X, y=None):
+        """Online computation of min and max on X for later scaling.
+        All of X is processed as a single batch. This is intended for cases
+        when `fit` is not feasible due dimensionality of X or because X is
+        read from a continuous stream.
+
+        Parameters
+        ----------
+        X : array-like with shape [n_samples_, n_features]
```

Comment:
nitpick: Can you modify this to `array-like, shape` to be consistent with documentation.